main.go

```diff
@@ -161,11 +163,21 @@ func main() {
 	monitoringServiceName := getStringEnv(monitoringServiceNameEnvironmentVariable, defaultMonitoringServiceName)
 
 	configID := uint8(getUintEnv(configurationIdEnvironmentVariable, 0))
-	config, err := ohttp.NewConfigFromSeed(configID, hpke.KEM_X25519_HKDF_SHA256, hpke.KDF_HKDF_SHA256, hpke.AEAD_AES128GCM, seed)
+	config, err := ohttp.NewConfigFromSeed(configID, hpke.KEM_X25519_KYBER768_DRAFT00, hpke.KDF_HKDF_SHA256, hpke.AEAD_AES128GCM, seed)
 	if err != nil {
 		log.Fatalf("Failed to create gateway configuration from seed: %s", err)
 	}
 
+	// From the primary configuration ID, create a key ID for the legacy configuration that old
+	// clients will use for obtaining configuration material. This will eventually be removed once all
+	// clients have been updated to support the primary configuration ID.
+	legacyConfigID := uint8((configID - 128) % 255)
+	seed[len(seed)-1] ^= 0xFF
```

Comment:
Why not the same seed?

Reply:
It feels wrong to use the same randomness for two different purposes, so I flipped some bits here.